collect_app/src/main/java/org/odk/collect/android/application/Collect.java

```diff
@@ -91,6 +92,9 @@
     @Inject
     public CollectJobCreator collectJobCreator;
 
+    @Inject
+    public GeneralSharedPreferences generalSharedPreferences;
```

Comment:
This should be using `MetaSharedPreferencesProvider` instead. That way we don't need to use `KEYS_WE_SHOULD_NOT_RESET`.

Reply:
Done.